public/app/plugins/datasource/cloud-monitoring/datasource.ts

```diff
@@ -179,7 +179,18 @@ export default class CloudMonitoringDatasource extends DataSourceWithBackend<
           const dataQueryResponse = toDataQueryResponse({
             data: data,
           });
-          return dataQueryResponse?.data[0]?.meta?.custom?.labels ?? {};
+          return _map(dataQueryResponse?.data, 'meta.custom.labels')
+            .filter((p) => !!p)
+            .reduce((acc, labels) => {
+              for (let key in labels) {
+                if (acc[key]) {
```

Comment:
you would need to avoid pushing labels that are already there:

![Screenshot from 2022-01-24 12-00-13](https://user-images.githubusercontent.com/4025665/150771019-adc0564f-b55c-44f2-82de-e78caa7aad09.png)

Reply:
That's good point. I didn't check multiple Group bys.
I fixed it in https://github.com/grafana/grafana/pull/44365/commits/1b4ed422719df6e70d3d0e9746fafd8ce08911af .